### PR TITLE
Update gcc version in build_toolchain.sh

### PR DESCRIPTION
**Pull request by nathan-tm-nguyen**
I encountered this error (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86724) using gcc 8.2.0 and had to manually add in the include statements in 'graphite.h' in order to successfully build the toolchain. Later versions of gcc fix this issue, so this error hopefully won't crop up again for other users. I also updated the versioning of the other dependencies, but feel free to reject this if it breaks backward compatibility somewhere.

### CODE REVIEW

**Comment by Vogtinator** [comment]
You bumped more than just the version of GCC - did you do a full rebulid of the SDK and ndless itself?
If that works, make sure to test the samples and some of the installers as well.

**Comment by nafanwin** [comment]
Hi @Vogtinator, sorry for the late reply. 

TLDR: executables created with gcc 8.3.0 don't work

I did a full rebuild of both ndless and the SDK with gcc 8.3.0. Everything was successfully built (no intervention on my part required), and I could compile all the provided samples. The installers were also compiled without any errors.

However, when I compared the checksums (SHA-256) of the calcbin installers generated with gcc 8.3.0 with the checksums of those generated with gcc 8.2.0, none of them matched. The same discrepancy also happened when I checked the executables generated from the samples. I tried testing the 4.5 installer generated with gcc 8.3.0 on a CX non-CAS handheld (HW - V)and on Firebird running CX CAS (HW - A), but neither worked; my handheld just restarted without installing ndless, and Firebird entered the debugger.

I'm not sure if this error message is helpful, but the Firebird debugger printed this after ndless failed to install:

> Warning (10338b90): Data abort: address=2f5a434e status=05 instruction at 10338b90

As for testing the newer versions of binutils, gdb, and newlib (tested separately from gcc 8.3.0), I didn't run into any issues except for newlib; for some reason, using newlib 3.1.0 creates the same errors mentioned in issue #141; I tried removing the .built file and attempted to rebuild, but the issue still persisted.

All in all, it's probably more trouble than its worth to bump gcc up to 8.3.0/update the other versions. I didn't think using newer versions would be this prone to error, so my apologies for not running thorough tests beforehand. If there's anything else I can do, please let me know.

**Comment by Vogtinator** [comment]
> I'm not sure if this error message is helpful, but the Firebird debugger printed this after ndless failed to install:
> 
> > Warning (10338b90): Data abort: address=2f5a434e status=05 instruction at 10338b90

That's the usual "4.5 installer is unreliable" issue. It usually fails after sending the installer file, but with a reboot in between it normally works.

> As for testing the newer versions of binutils, gdb, and newlib (tested separately from gcc 8.3.0), I didn't run into any issues except for newlib; for some reason, using newlib 3.1.0 creates the same errors mentioned in issue #141; I tried removing the .built file and attempted to rebuild, but the issue still persisted.

newlib 3.1.0 might've removed some symbols, so nSDL might need a rebuild.

**Comment by nafanwin** [comment]
I tried the 4.5 installer generated by 8.3 on both of my handhelds (non-CAS HW-V and CAS HW-AC) with reboots in between attempts; still no luck (over 5 attempts per calc). 

As for nSDL, I'll try to rebuild it when I can find some time.

**Comment by Vogtinator** [comment]
> I tried the 4.5 installer generated by 8.3 on both of my handhelds (non-CAS HW-V and CAS HW-AC) with reboots in between attempts; still no luck (over 5 attempts per calc).

Ok, so a diff of the object/binary files would be helpful here.

**Comment by nafanwin** [comment]
I compared the distributed 4.5 installer to the 8.3 generated installer, and it shows that 8.3 is mostly the same except for lines 23-24 and 66437-69109. Here's a side-by-side comparison in hex (original on left, 8.3 on right): https://filebin.ca/4a3cGUZNcT1J/diff.txt

**Comment by Vogtinator** [comment]
Diffing the .tns won't help much, you should diff the intermediate files, (.o, .bin, etc.).

**Comment by Vogtinator** [comment]
I'll close this for now as #211 bumped it even further, though it omitted newlib 3.1.0.
